File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/repository/TjvOrderRepository.java
package wrzecond.repository;

import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import wrzecond.entity.TjvOrder;
import wrzecond.entity.TjvTable;

import java.util.List;

@Repository
public interface TjvOrderRepository extends TjvRepository<TjvOrder> {

    @Query("SELECT o FROM TjvOrder o WHERE o.table = :table AND o.completed = 0")
    List<TjvOrder> findByTableID (TjvTable table);

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/repository/TjvOrderRepository.kt
package wrzecond.repository

import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import wrzecond.entity.TjvOrder
import wrzecond.entity.TjvTable

@Repository
interface TjvOrderRepository : TjvRepository<TjvOrder> {

    @Query("SELECT o FROM TjvOrder o WHERE o.table = :table AND o.completed = false")
    fun findByTableID (table: TjvTable) : List<TjvOrder>

}
